src/pageServices/contentScripts/highlight.js

```diff
@@ -287,24 +291,39 @@ export const highlightOnPage = () => {
     events.forEach((eventName) => {
       document.removeEventListener(eventName, scrollListenerCallback, true);
     });
-    document.removeEventListener("click", clickListener);
+
+    document.removeEventListener("dblclick", onElementDblClick);
+
     listenersAreSet = false;
   };
 
   const removeHighlight = (callback) => () => {
     removeEventListeners(removeHighlightElements(callback));
   };
 
-  const clickListener = (event) => {
-    if (!event.clientX && !event.clientY) return;
+  const onElementDblClick = (evt) => {
+    const elements = predictedElements?.filter((element) => element.jdnHash === evt.target.id);
+    const notAddedToPO = () => elements.some(({ generate }) => !generate);
```

Comment:
this code looks unnecessarily complex. no need to use Array methods to evaluate what you already 've got on 305 line.